packages/app/hooks/useAlgoliaSearch.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from "react";
+import { useDebounce } from "app/hooks/useDebounce";
+import { SearchIndex } from "algoliasearch/lite";
+
+type Props = {
+  searchPhrase: string;
+  algoliaIndex: SearchIndex;
+};
+
+type SearchResult<T> = {
+  loading: boolean;
+  data: T[];
+};
+
+export function useAlgoliaSearch<T>({
+  searchPhrase,
+  algoliaIndex,
+}: Props): SearchResult<T> {
+  const [searchResult, setSearchResult] = useState<T[]>([]);
+  const debouncedSearchPhrase = useDebounce(searchPhrase, 200);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+  }, [searchPhrase]);
+
+  useEffect(() => {
+    if (!debouncedSearchPhrase) return;
+
+    const search = async () => {
+      try {
+        const algoliaSearchResult = await algoliaIndex.search<T>(
+          debouncedSearchPhrase
+        );
+        setSearchResult(algoliaSearchResult.hits);
+      } catch (ex) {
+        console.error(ex);
```

Comment:
we should do something about that error. Let's add it to the `SearchResult` and display some error message i.ex. `Something went wrong. Try again.`.